py/mpprint.h

```diff
@@ -80,3 +80,59 @@ int mp_vprintf(const mp_print_t *print, const char *fmt, va_list args);
 #endif
 
 #endif // MICROPY_INCLUDED_PY_MPPRINT_H
+
+// You can use several MP_PRN_LEVEL defines in separate _.c files
+#if defined(MP_PRN_LEVEL) && (MP_PRN_LEVEL > 0)
+// Debug messages during code developing with MP_PRN(level, ...) & MP_PRN_LEVEL.
+// An approximate hierarchy of debug levels MP_PRN_LEVEL is:
+#define MP_PRN_SUPPRESS 0 // SUPPRESS all messages. Use it in the release version.
+#define MP_PRN_CRITICAL 1 // For the most CRITICAL errors, often requiring a system reset. Use a message with this level, if possible, raising an exception.
+#define MP_PRN_ERROR 2 // ERROR requiring program restart, use message with this level before raising an exception.
+#define MP_PRN_WARNING 3 // WARNING, something went wrong, but you can fix it with additional operations in code right now or may ignore it.
+#define MP_PRN_INFO 4 // INFO, it is interesting and useful for understanding a bug.
+#define MP_PRN_DEBUG 5 // DEBUG, more detailed information, dig deeper.
+#define MP_PRN_TRACE 6 // TRACE, show a flow of the algorithm, like enter/exit a function.
```

Comment:
These defines would need to be above the #ifdef so they can be used to define the desired level

Reply:
Moved upper